lib/wallet/sync.rs

```diff
@@ -87,53 +93,6 @@ impl WalletInner {
             "applying block to BDK wallet"
         );
 
-        let bdk_tip_height = wallet_write.local_chain().tip().height();
-
-        // TODO: should really run the entire handle_connect_block logic here
-        if bdk_tip_height < block_height - 1 {
-            tracing::info!(
-                bdk_tip_height = bdk_tip_height,
-                block_height = block_height,
-                "BDK tip is behind newly received block, catching up"
-            );
```

Comment:
It's not clear to me why this is not working. On master, catching up multiple blocks at once in the sync leads to an error. On this PR, it works. Is it related to the `with_mut` stuff? I have to admit it's a bit over my head

Reply:
We're doing one commit for each block now, vs. only doing one commit for the entire batch. A single commit would be best - but not when it's not working. Probably OK for now, IMO.